### DISCUSSION TITLE
Demarcation of Power Ports - devices connected to Power Feeds that have physical outlets

### DISCUSSION THREAD

**Discussion by bluikko** [netbox-community/netbox#7837] [Q&A]
Encountered an interesting issue modeling Power Ports.

Some devices (device "A") have a cable permanently attached to them (a "hardwired" Power Port) and of course at the other end there is a plug that connects to Power Outlet of another device.
Modeling this is straightforward if device A connects to another device:
- Device A with a Power Port: Type Hardwired -> Cable -> Device B with a Power Outlet: Type say C19.
This models the situation correctly, a cable is hardwired to Device A and connects with C-type connector to Device B.

What happens when such a device connects to a Power Feed that currently do not have connector types at all?
Then it becomes:
- Device A Power Port: Type Hardwired -> Cable -> Power Feed.
This looks like there is an all-the-way hardwired connection from the device to a Power Feed. But this is not true - the Power Feed does have an outlet (in my case they all do), say IEC60309 type.

Am I missing something here?

For example: should there actually be "dummy" or "outlet" "devices" for each Power Feed that models the actual physical power outlet of the feed? That seems like the currently only way to correctly model Power Feed that has an outlet connector.

Are other users modeling Power Feeds using such an "outlet device" and is that how this should be done? Has there been thoughts about giving Power Feeds a connector so that it could have a type (it seems a bit complex to create devices for each outlet of a feed)?

**Comment by cpmills1975** [comment]
Given the power port is usually a male pinned something, in the case you are thinking about, I'd just consider the power port to be physically on the end of the attached cable.

The way I use the power port type is to know what type of cable and reciprocal socket I need to connect a device to. Having a device with a power port that's technically the wrong end of the cable isn't so wrong. If it's an IEC60309 type plug, it's still going to need a matching socket to connect it to on the feed end.

**Reply by bluikko** [reply]
This was the second option that I was thinking, and had actually used myself. But other team members had started to mark permanently attached cables as "hardwired" which lead to my OP - I hadn't even thought about this before.

In this option, the Cable between such a device and another device would become a "zero-length" cable so I can see how it would confuse some.

Would be interesting to hear if this is the popular way to do it.
